src/gpio_monitor.cpp

```diff
@@ -1,31 +1,132 @@
 #include "gpio_monitor.hpp"
 
+#include "constants.hpp"
 #include "logger.hpp"
 #include "types.hpp"
+#include "utility/dbus_utility.hpp"
 #include "utility/json_utility.hpp"
 
+#include <boost/asio.hpp>
+#include <boost/bind/bind.hpp>
+#include <gpiod.hpp>
+
 namespace vpd
 {
 bool GpioEventHandler::getPresencePinValue()
 {
-    // ToDo Add implementation.
-    return true;
+    gpiod::line l_presenceLine = gpiod::find_line(m_presencePin);
+
+    if (!l_presenceLine)
+    {
+        logging::logMessage("Couldn't find presence line: " + m_presencePin);
+        return m_prevPresencePinValue;
+    }
+
+    l_presenceLine.request(
+        {"Presnce line", gpiod::line_request::DIRECTION_INPUT, 0});
+    bool l_presencePinValue = l_presenceLine.get_value();
+
+    return l_presencePinValue;
 }
 
 void GpioEventHandler::toggleGpio()
 {
-    // ToDo Add implementation.
+    bool l_isFruPresent = false;
```

Comment:
Though this is outside the scope of this change, but after going through the implementation, I think now the name of this API does not justify the body.
We will have to come up with a new name for this API.

Reply:
Updated